File: 卡码/代码随想录/1/06-二叉树/Python/02-二叉树属性/LeetCode_0104.py
#104. 二叉树的最大深度
from typing import *
from ..TreeNode import *

class Solution:
  def maxDepth(self, root: Optional[TreeNode]) -> int:
    if not root:
      return 0
    return 1+max(self.maxDepth(root.left),self.maxDepth(root.right))

File: 卡码/代码随想录/1/06-二叉树/Go/02-二叉树属性/LeetCode_0104.go
//104. 二叉树的最大深度
package main

import (
	. "binaryTree/types"
)

func max(a,b int) int{
	if a>b {
		return a
	}else{
		return b
	}
}

func maxDepth(root *TreeNode) int {
	if root==nil {
		return 0
	}
	return 1+max(maxDepth(root.Left),maxDepth(root.Right))
}
